packages/cli/src/dev.ts

```diff
@@ -0,0 +1,143 @@
+import Ajv, { Schema } from 'ajv';
+import addFormats from 'ajv-formats';
+import chokidar from 'chokidar';
+import * as fs from 'fs';
+import * as yaml from 'js-yaml';
+import { Miniflare } from 'miniflare';
+import ora from 'ora';
+import * as path from 'path';
+
+import { buildScriptFromManifest } from './build';
+import { prettyPath } from './files';
+import { getDefaultManifestPath, resolveFile, resolveIntegrationManifestPath } from './manifest';
+import { getAPIClient } from './remote';
+import { createDevTunnel } from './tunnel';
+
+export const GITBOOK_DEV_SERVER_PORT = 8787;
+export const GITBOOK_DEV_CONFIG_FILE = 'gitbook-dev.yaml';
+
+const spinner = ora({ color: 'blue' });
+
+/**
+ * Start the integrations dev server on the given port. If no port is given,
+ * the default one is used. The dev server will automatically reload changes to
+ * the integration script.
+ */
+export async function startIntegrationsDevServer(port: number = GITBOOK_DEV_SERVER_PORT) {
+    spinner.start('Starting dev server...');
+    /**
+     * Read the integration manifest and build the integration script so that
+     * it can be served by the dev server.
+     * Also, read the dev config file to get the space to use for the dev server.
```

Comment:
Can it be specified by the cli arguments instead?

Having to write my space in the manifest feels slow. Maybe `gitbook dev <spaceId>` could write it for me.

Reply:
Also you don't seem to handle the case the space is not defined to show a proper error to the user.

Reply:
I'm more inclined on not having it as a CLI option but use a `gitbook-dev.yaml` config to specify with easy to support it in the future too

Reply:
> Also you don't seem to handle the case the space is not defined to show a proper error to the user.

The error would be thrown by the API when we are setting the tunnel if `space` does not exist. Would that not be enough? 😁

Reply:
> The error would be thrown by the API when we are setting the tunnel if space does not exist. Would that not be enough? 😁

Yes, I was referring to the fact that the space isn ot even defined (empty string, undefined).

> I'm more inclined on not having it as a CLI option but use a gitbook-dev.yaml config to specify with easy to support it in the future too

I disagree, I think it adds friction, and makes it harder to test on multiple spaces (why would we want the user to use only one space for an integration?).

Reply:
> Yes, I was referring to the fact that the space isn ot even defined (empty string, undefined).
Oh hm, I thought it would be taken care of by `Ajv` when we validate the dev config yaml but I'll add another check in code

> I disagree, I think it adds friction, and makes it harder to test on multiple spaces (why would we want the user to use only one space for an integration?).
You can still test on multiple spaces. All it needs is a change in the dev config `space: newSpaceId`. 

Maybe I misunderstood your first question. We certainly can generate an empty `.gitbook-dev.yaml` during `gitbook create` but I'm not sure about polluting the command with CLI args because it can easily blow if this dev config was to expand